analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/AnalysisScaffolding.cs

```diff
@@ -80,6 +81,42 @@ public static string CreateSonarProjectConfigWithFilesToAnalyze(TestContext cont
         public static string CreateSonarProjectConfig(TestContext context, ProjectType projectType, bool isScannerRun = true) =>
             CreateSonarProjectConfig(context, "ProjectType", projectType.ToString(), isScannerRun);
 
+        public static string CreateSonarLintXml(
+            TestContext context,
+            string language = LanguageNames.CSharp,
+            bool analyzeGeneratedCode = false,
+            string[] exclusions = null,
+            string[] inclusions = null,
+            string[] globalExclusions = null,
+            string[] testExclusions = null,
+            string[] testInclusions = null,
+            string[] globalTestExclusions = null) =>
+            TestHelper.WriteFile(context, "SonarLint.xml", GenerateSonarLintXmlContent(language, analyzeGeneratedCode, exclusions, inclusions, globalExclusions, testExclusions, testInclusions, globalTestExclusions));
+
+        public static string GenerateSonarLintXmlContent(
+            string language = LanguageNames.CSharp,
+            bool analyzeGeneratedCode = false,
+            string[] exclusions = null,
+            string[] inclusions = null,
+            string[] globalExclusions = null,
+            string[] testExclusions = null,
+            string[] testInclusions = null,
+            string[] globalTestExclusions = null) =>
+            new XDocument(
+                new XDeclaration("1.0", "utf-8", "yes"),
+                new XElement("AnalysisInput",
+                    new XElement("Settings",
+                        CreateKeyValuePair("Setting", $"sonar.{(language == LanguageNames.CSharp ? "cs" : "vbnet")}.analyzeGeneratedCode", analyzeGeneratedCode.ToString()),
+                        CreateKeyValuePair("Setting", "sonar.exclusions", string.Join(",", exclusions ?? Array.Empty<string>())),
```

Comment:
Instead of repeating `string.Join(",", exclusions ?? Array.Empty<string>())` For each parameter. You can create a method that received the string array and returns you a string. and call that method for all parameters. It should reduce the codebase a bit.